### DISCUSSION TITLE
Unchanged value of cart.estimatedCost.totalAmount after cartDiscountCodesUpdate

### DISCUSSION THREAD

**Discussion by bannzai** [Shopify/storefront-api-feedback#87] [Help]
## Abstract
My app can apply discount to cart with discount code via cartDiscountCodesUpdate and want to update screen with cartDiscountCodesUpdate response. cartDiscountCodesUpdate mutation is successfully, But CartDiscountCodesUpdatePayload#cart.estimatedCost.totalAmout is not changed compare of before mutation.

[cartDiscountCodesUpdate#DiscountCodesUpdatePayload](https://shopify.dev/api/storefront/2022-01/objects/Cart) is written about returning value of updated cart in document.

> The updated cart.

Please help me to confirm and response about this discussion . Thank you!

**Comment by johncraigcole** [comment]
Hey @bannzai the `totalAmount` should be updated on the response to reflect the new total with the applied discount code. 

The only reasons the discount code won't update the `totalAmount` is if the discount code is not active or the contents of the cart don't satisfy the requirements of the discount. In my own testing I've also made the mistake of having a typo in my discount code 😄.

**Reply by yuzushioh** [reply]
It looks like this bug was already fixed by shopify and it works fine now 😊

**Comment by zwergius** [comment]
I am seeing this issue on 2022-04 api 
<img width="642" alt="image" src="https://user-images.githubusercontent.com/2060308/162166300-e8fd2d44-f324-48f0-9a7f-e0125023c55b.png">

totalAmount makes no sense at all.

Discount code applied is 50% this is correctly reflected when I redirect to the checkoutUrl

**Comment by LiahMartens** [comment]
I ran into this exact same issue. If you apply a discount code and then remove it by passing an empty array to the `cartDiscountCodesUpdate` mutation the totalAmount will not update for some reason.

My workaround is to not send an empty array but rather send an empty discount code as the `discountCodes` argument.
Ie: `cartDiscountCodesUpdate(cartId: $cartId, discountCodes: [''])` 

This forces storefront to recalculate the line item prices because it is seen as a new discount code. The only caveat is that this empty discount code would also show up in the discounts array being returned in the cart object but it's  easy enough to filter out.

@johncraigcole I believe this is in fact a bug as a discount code removal should trigger an update of the line item prices.
